include/wabt/common.h

```diff
@@ -297,9 +304,162 @@ enum class RelocType {
 
   First = FuncIndexLEB,
   Last = FuncIndexI32,
+  None = -1, // Used internally as a sentinel value
+};
+
+enum class RelocDataType {
+  I32, I64,
+  LEB, LEB64,
+  SLEB, SLEB64,
+};
+
+constexpr size_t kRelocDataTypeSize[] {
+  4, 8,
+  5, 10,
+  5, 10
+};
+
+constexpr RelocDataType kRelocDataType[] {
+  RelocDataType::LEB,    // FuncIndexLEB = 0
+  RelocDataType::SLEB,   // TableIndexSLEB = 1
+  RelocDataType::I32,    // TableIndexI32 = 2
+  RelocDataType::LEB,    // MemoryAddressLEB = 3
+  RelocDataType::SLEB,   // MemoryAddressSLEB = 4
+  RelocDataType::I32,    // MemoryAddressI32 = 5
+  RelocDataType::LEB,    // TypeIndexLEB = 6
+  RelocDataType::LEB,    // GlobalIndexLEB = 7
+  RelocDataType::I32,    // FunctionOffsetI32 = 8
+  RelocDataType::I32,    // SectionOffsetI32 = 9
+  RelocDataType::LEB,    // TagIndexLEB = 10
+  RelocDataType::SLEB,   // MemoryAddressRelSLEB = 11
+  RelocDataType::SLEB,   // TableIndexRelSLEB = 12
+  RelocDataType::I32,    // GlobalIndexI32 = 13
+  RelocDataType::LEB64,  // MemoryAddressLEB64 = 14
+  RelocDataType::SLEB64, // MemoryAddressSLEB64 = 15
+  RelocDataType::I64,    // MemoryAddressI64 = 16
+  RelocDataType::SLEB64, // MemoryAddressRelSLEB64 = 17
+  RelocDataType::SLEB64, // TableIndexSLEB64 = 18
+  RelocDataType::I64,    // TableIndexI64 = 19
+  RelocDataType::LEB,    // TableNumberLEB = 20
+  RelocDataType::SLEB,   // MemoryAddressTLSSLEB = 21
+  RelocDataType::I64,    // FunctionOffsetI64 = 22
+  RelocDataType::I32,    // MemoryAddressLocRelI32 = 23
+  RelocDataType::SLEB64, // TableIndexRelSLEB64 = 24
+  RelocDataType::SLEB64, // MemoryAddressTLSSLEB64 = 25
+  RelocDataType::I32,    // FuncIndexI32 = 26
+};
+
+enum class RelocKind {
```

Comment:
So RelocKind is a wabt-internal thing, that tried to group the different types of relocations defined in the spec?     Is is this kind also defined upstream?

Maybe worth a comment here.